src/SSHDebugPS/Docker/DockerHelper.cs

```diff
@@ -17,6 +17,87 @@ public class DockerHelper
         private const string dockerPSCommand = "ps";
         // --no-trunc avoids parameter truncation
         private const string dockerPSArgs = "-f status=running --no-trunc --format \"{{json .}}\"";
+        private const string dockerInspectCommand = "inspect";
+        private const string dockerInspectArgs = "-f \"{{json .Platform}}\" ";
+
+        public static string GetContainerPlatform(string hostname, string containerName)
+        {
+            string containerPlatform = string.Empty;
+
+            DockerCommandSettings settings = new DockerCommandSettings(hostname, false);
+            settings.SetCommand(dockerInspectCommand, string.Concat(dockerInspectArgs, containerName));
+
+            LocalCommandRunner commandRunner = new LocalCommandRunner(settings);
+
+            StringBuilder errorSB = new StringBuilder();
+            int? exitCode = null;
+
+            try
+            {
+                ManualResetEvent resetEvent = new ManualResetEvent(false);
+                commandRunner.ErrorOccured += ((sender, args) =>
+                {
+                    if (!string.IsNullOrWhiteSpace(args.ErrorMessage))
+                    {
+                        errorSB.AppendLine(args.ErrorMessage);
+                    }
+                    resetEvent.Set();
+                });
+
+                commandRunner.Closed += ((sender, args) =>
+                {
+                    exitCode = args;
+                    resetEvent.Set();
+                });
+
+                commandRunner.OutputReceived += ((sender, args) =>
+                {
+                    if (!string.IsNullOrWhiteSpace(args))
+                    {
+                        containerPlatform = args.Trim().Replace("\"", "").Equals("windows") ? "Windows" : "Linux";
+                    }
+                });
+
+                commandRunner.Start();
+
+                bool cancellationRequested = false;
+                VS.VSOperationWaiter.Wait(UIResources.RetrievingContainerPlatformsMessage, false, (cancellationToken) =>
+                {
+                    while (!resetEvent.WaitOne(2000) && !cancellationToken.IsCancellationRequested)
+                    { }
+                    cancellationRequested = cancellationToken.IsCancellationRequested;
+                });
+
+                if (!cancellationRequested)
+                {
+                    // might need to throw an exception here too??
+                    if (exitCode.GetValueOrDefault(-1) != 0)
+                    {
+                        // if the exit code is not zero, then the output we received possibly is the error message
```

Comment:
Do we want to handle Telemetry here to see what invalid messages we get?

```
HostTelemetry.SendEvent(Telemetry.Event.Error_DockerPlatform, new KeyValuePair<string, object>[] {
      new KeyValuePair<string, object>(DockerPlatform_Error, errorSB.ToString())
}); 
```

Add `Error_DockerPlatform` in https://github.com/microsoft/MIEngine/blob/a95f162a88070c323c4976146a1ae16250c4270f/src/SSHDebugPS/Utilities/Telemetry.cs#L8